### PR TITLE
III-5875 Exclude param for public route

### PR DESCRIPTION
**Pull request by grubolsch**
### Added
Added a configuration parameter to mark which routes are always requiring authentication, or make it toggleable dependent on a GET param.

---
Ticket: https://jira.uitdatabank.be/browse/III-5875

### CODE REVIEW

File: app/Authentication/AuthServiceProvider.php [hunk 1]
```diff
@@ -59,15 +59,14 @@ function () use ($container): RequestAuthenticatorMiddleware {
...
-                    $container->get('config')['authenticate_public_routes'] ?? false
                 );
 
                 // We can not expect the ids of events, places and organizers to be correctly formatted as UUIDs,
                 // because there is no exhaustive documentation about how this is handled in UDB2. Therefore we take a
                 // rather liberal approach and allow all alphanumeric characters and a dash as ids.
                 $authenticator->addPublicRoute('~^/offers/?$~', ['GET']);
                 $authenticator->addPublicRoute('~^/(events?|places?)/?$~', ['GET']);
-                $authenticator->addPublicRoute('~^/(events?|places?)/[\w\-]+/?$~', ['GET']);
+                $authenticator->addPublicRoute('~^/(events?|places?)/[\w\-]+/.*?$~', ['GET'], 'embedContributors');
```
Comment (LucWollants) [inline app/Authentication/AuthServiceProvider.php:69]: It seems that there is a small change to this regex, is that needed?

File: src/Http/Auth/RouteRule.php [hunk 2]
```diff
@@ -4,23 +4,37 @@
...
 
-    public function __construct(string $pathPattern, array $methods)
+    public function __construct(string $pathPattern, array $methods, ?string $excludeGetParams = null)
     {
         $this->pathPattern = $pathPattern;
         $this->methods = $methods;
+        $this->excludeQueryParam = $excludeGetParams;
     }
 
     public function matchesRequest(ServerRequestInterface $request): bool
```
Comment (LucWollants) [inline src/Http/Auth/RouteRule.php:23]: With this extra change a unit test could be very useful.

File: src/Http/Auth/RequestAuthenticatorMiddleware.php [hunk 3]
```diff
@@ -43,29 +43,26 @@ final class RequestAuthenticatorMiddleware implements MiddlewareInterface
...
-        bool $authenticatePublicRoutes
+        UserPermissionsReadRepositoryInterface $userPermissionsReadRepository
     ) {
         $this->uitIdV1JwtValidator = $uitIdV1JwtValidator;
         $this->uitIdV2JwtValidator = $uitIdV2JwtValidator;
         $this->apiKeyAuthenticator = $apiKeyAuthenticator;
         $this->apiKeyConsumerReadRepository = $apiKeyConsumerReadRepository;
         $this->apiKeyConsumerPermissionCheck = $apiKeyConsumerPermissionCheck;
         $this->userPermissionReadRepository = $userPermissionsReadRepository;
-        $this->authenticatePublicRoutes = $authenticatePublicRoutes;
```
Comment (LucWollants) [inline src/Http/Auth/RequestAuthenticatorMiddleware.php:63]: Don't forget to make a PR to remove this config

**Review by LucWollants — CHANGES_REQUESTED** [review]
Some extra changes are needed.

File: app/Authentication/AuthServiceProvider.php (hunk 1 above)
Comment (grubolsch) [reply app/Authentication/AuthServiceProvider.php]: no I have restored it

File: src/Http/Auth/RequestAuthenticatorMiddleware.php (hunk 3 above)
Comment (grubolsch) [reply src/Http/Auth/RequestAuthenticatorMiddleware.php:63]: https://github.com/cultuurnet/appconfig/pull/486

File: src/Http/Auth/RouteRule.php (hunk 2 above)
Comment (grubolsch) [reply src/Http/Auth/RouteRule.php:23]: See PublicRouteRuleTest

**Comment by grubolsch** [comment]
Feedback processed, also added related PR https://github.com/cultuurnet/appconfig/pull/486

**Review by simon-debruijn — APPROVED** [review]

**Review by LucWollants — APPROVED** [review]

**Pull request merged**